lib/driver.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'csv_record'
+
+
+module RideShare
+  class Driver < CsvRecord
+    VALID_STATUSES = [:AVAILABLE , :UNAVAILABLE]
+
+    attr_reader :id, :name, :vin, :status, :trips
+    attr_writer :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: [])
+      super(id)
+
+      validate_status(status)
+      @status = status
+      @name = name
+      validate_vin(vin)
+      @vin = vin
+      @trips = trips
+    end
+
+    def validate_vin(vin)
+      if vin.length != 17
+        raise ArgumentError.new("Invalid vin. Must be 17 digits.")
+      end
+    end
```

Comment:
Love the separated-out validation methods! so thorough, without cluttering up the constructor :)